apidef/oas/default.go

```diff
@@ -7,18 +7,22 @@ import (
 )
 
 const (
-	invalidServerURLFmt = "Please update %q to be a valid url or pass a valid url with upstreamURL query param"
+	invalidServerURLFmt          = "Please update %q to be a valid url or pass a valid url with upstreamURL query param"
+	unsupportedSecuritySchemeFmt = "unsupported security scheme: %s"
 )
 
 var (
 	errEmptyServersObject = errors.New("servers object is empty in OAS")
 	errInvalidUpstreamURL = errors.New("invalid upstream URL")
 	errInvalidServerURL   = errors.New("Error validating servers entry in OAS")
+
+	errEmptySecurityObject = errors.New("security object is empty in OAS")
 )
 
 type TykExtensionConfigParams struct {
-	UpstreamURL string
-	ListenPath  string
+	UpstreamURL    string
+	ListenPath     string
+	Authentication bool
```

Comment:
shall we keep `Authentication` flag a ptr `*bool`, thereby we can decide whether the query parameter was not supplied(nil). Both `false` and `true` can be a valid value for `authentication` query param, hence we cannot rely on boolean default value `false`

Reply:
Added